src/test/java/com/google/step/youtube/YouTubeInfoScraperTest.java

```diff
@@ -310,4 +313,57 @@ public void scrapeVideoIdsFromSearch_IOException() throws IOException {
         assertThrows(IOException.class, () -> scraper.scrapeVideoIdsFromSearch(VIDEO_ID));
     }
 
+    @Test
+    public void scrapePromoCodesFromVideos_emptyVideoIdList() throws IOException {
+        VideoListResponse testVideoResponse = new VideoListResponse();
+        when(mockListVideos.execute()).thenReturn(testVideoResponse);
+        Optional<List<PromoCode>> actual = scraper.scrapePromoCodesFromVideos(KEYWORD, EMPTY_VIDEO_ID_LIST);
+        assertThat(actual.isPresent(), equalTo(false));
+    }
+
+    /** Multiple items with one item having no promo-code in the description. */
```

Comment:
nit: Does this need to be a javadoc comment? Since this is about implementation rather than documentation.

Reply:
Fixed.